### PR TITLE
Add function parser for implicit definition in tuple

### PR DESCRIPTION
**Pull request by MortalFlesh**

### CODE REVIEW

File: src/Service/Functions.php [hunk 1]
```diff
@@ -102,4 +102,22 @@ public function getFunctionNamesByParameter(string $possiblyParameter): iterable
             }
         }
     }
+
+    public function getFunctionNamesByAllParameters(array $possiblyParameters): iterable
```
Comment (legendik) [inline src/Service/Functions.php]: `$possiblyParameters` -> `$possibleParameters`.

Btw, better will be `$allParameters`, like the function name 🤔

File: src/Service/Functions.php [hunk 2]
```diff
@@ -102,4 +102,22 @@ public function getFunctionNamesByParameter(string $possiblyParameter): iterable
...
+        $sortedPossibleParameters = $this->sort($possiblyParameters);
+
+        foreach ($this->functionParameters as $functionName => $parameters) {
+            if ($this->sort($parameters) === $sortedPossibleParameters) {
+                yield $functionName;
+            }
+        }
+    }
+
+    private function sort(array $array): array
```
Comment (legendik) [inline src/Service/Functions.php:117]: Why own private function without any other meaning when you can call `sort` directly? 🤔

File: src/Service/Parser/FunctionParser/ImplicitFunctionDefinitionByTupleParser.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+class ImplicitFunctionDefinitionByTupleParser extends AbstractFunctionParser
+{
+    /**
+     * @param string|array $rawValue Raw value from query parameters
+     */
+    protected function supportsParameters(array $queryParameters, string $rawColumn, $rawValue): bool
+    {
+        if ($this->isTuple($rawColumn)) {
+            $possiblyParameters = Tuple::parse($rawColumn)->toArray();
```
Comment (legendik) [inline src/Service/Parser/FunctionParser/ImplicitFunctionDefinitionByTupleParser.php]: `$possibleParameters`

**Review by legendik — CHANGES_REQUESTED** [review]

File: src/Service/Functions.php (hunk 2 above)
Comment (MortalFlesh) [reply src/Service/Functions.php:117]: because `sort` is sorting by reference and cannot be chained

**Comment by danger-lmc** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>There are some fixup/squash commits that needs to be applied before merge.</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c6c86f3d1870ea3f6841960e8e3f87e5d3cdfc51
</p>

**Review by legendik — APPROVED** [review]

**Review by legendik — APPROVED** [review]

**Pull request merged**